core/src/Streamly/Internal/Data/Array/Type.hs

```diff
@@ -802,14 +858,29 @@ fromByteStr (Ptr addr#) = fromByteStr# addr#
 -- that is double the size of the array stream.
 --
 {-# INLINE fromChunkStreamK #-}
-fromChunkStreamK :: (Unbox a, MonadIO m) => StreamK m (Array a) -> m (Array a)
+fromChunkStreamK :: (MonadIO m, Unbox a) => StreamK m (Array a) -> m (Array a)
 fromChunkStreamK stream =
+    -- We buffer the entire stream and then allocate the target array of the
+    -- same size, thus requiring double the memory.
     fmap unsafeFreeze $ MA.fromChunkStreamK $ fmap unsafeThaw stream
 
 {-# DEPRECATED fromArrayStreamK "Please use fromChunkStreamK instead." #-}
 fromArrayStreamK :: (Unbox a, MonadIO m) => StreamK m (Array a) -> m (Array a)
 fromArrayStreamK = fromChunkStreamK
 
+-- | Given a stream of arrays, splice them all together to generate a single
+-- array. The stream must be /finite/.
+--
+{-# INLINE fromChunks #-}
+fromChunks :: (MonadIO m, Unbox a) => Stream m (Array a) -> m (Array a)
```

Comment:
`fromChunkStream`? Similar to `fromChunkStreamK`?
Or rename `fromChunkStreamK` to `fromChunksStreamK` or `fromChunksK`?

Reply:
`fromChunkStreamK` will be renamed to `fromChunksK`.